src/components/posts/Comments.js

```diff
@@ -0,0 +1,82 @@
+import React, { useEffect, useState } from 'react'
+import CommentItem from './CommentItem'
+import FilterMenu from './utils/filterMenu'
+
+const Comments = ({ post, setUser }) => {
+  const [sortType, setSortType] = useState('Most Recent')
+  const [comments, setComments] = useState(post.comments)
+  const [showingAll, setShowingAll] = useState(false);
+
+  useEffect(() => {
+    setComments(post.comments)
+    handleSorting()
+    // eslint-disable-next-line
+  }, [sortType, post])
+
+  const handleSorting = () => {
+    if (sortType === 'Most Recent') {
+      const sortedComments = [...post.comments]
+      setComments(sortedComments.sort((a, b) => {
+        return new Date(b.createdAt) - new Date(a.createdAt)
+      }));
+    }
+    if (sortType === 'Oldest') {
+      const sortedComments = [...post.comments]
+      setComments(sortedComments.sort((a, b) => {
+        return new Date(a.createdAt) - new Date(b.createdAt)
+      }));
+    }
+  }
+
+  const handleShowAll = () => {
+    if (showingAll) {
+      setShowingAll(false)
+    }
+    if (!showingAll) {
+      setShowingAll(true)
+    }
```

Comment:
What you have here is a boolean switch. This could be simplified into a single line by just setting `showingAll` to the opposite of what it currently is:

```js
setShowingAll(!showingAll)
```